tests/codegen/equivalent-slice-impls.rs

```diff
@@ -0,0 +1,23 @@
+// This test checks that two equivalent implementations produce the same number of `if` instructions.
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @f1
+// CHECK: if
+pub fn f1(input: &mut &[u64]) -> Option<u64> {
+  match input {
+      [] => None,
+      [first, rest @ ..] => {
+          *input = rest;
+          Some(*first)
+      }
+  }
+}
+
+// CHECK-LABEL: @f2
+// CHECK: if
```

Comment:
I would not have expected this to be enough to check that there are equal numbers of if instructions... don't need something that checks there's not a subsequent `if` too?